tools/wptserve/tests/functional/test_request.py

```diff
@@ -113,6 +114,17 @@ def handler(request, response):
         resp = self.request("/test/some_route")
         self.assertEqual(b"some route", resp.read())
 
+    def test_unicode_in_headers(self):
+        @wptserve.handlers.handler
+        def handler(request, response):
+            return request.headers["foo"]
+
+        route = ("GET", "/test/test_unicode_in_headers", handler)
+        self.server.router.register(*route)
+        encoded_text = u"你好".encode("utf-8")
```

Comment:
I think we should test more than just UTF-8 (given this would still pass with `header.decode('utf-8')`).

Reply:
I'd rather this be done with `pytest.mark.parametrize` so they run as separate tests, FWIW.

Reply:
This is a neat feature! However, it looks like `pytest.mark.parametrize` [does not work](https://docs.pytest.org/en/latest/unittest.html#pytest-features-in-unittest-testcase-subclasses) with `unittest.TestCase`.

Reply:
Oh, I missed your response here!

Ergh, another reason to move away from `unittest` entirely, but that probably shouldn't block this PR.